File: Prac3_Island_Builder/app/src/main/java/com/example/prac3_island_builder/MainActivity.java
package com.example.prac3_island_builder;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    ArrayList<Structure> selectorData;
    MapElement[][] mapGridData;

    public MainActivity() {
        //adds selector data to an arrayList
        selectorData = new ArrayList<>();
        StructureData sd = StructureData.get();
        for (int i = 0; i < sd.size(); i++) {
            selectorData.add(sd.get(i));
        }
        //gets map data fills 2D array
        MapData md = MapData.get();
        mapGridData = new MapElement[MapData.HEIGHT][MapData.WIDTH];

        for (int i = 0; i < MapData.HEIGHT; i++) {
            for (int j = 0; j < MapData.WIDTH; j++) {
                mapGridData[i][j] = md.get(i,j);
            }
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        FragmentManager frag = getSupportFragmentManager();

       //instantiates selector list
        Selector selectorFrag = (Selector) frag.findFragmentById(R.id.selectorFrame);
        if (selectorFrag == null){
            selectorFrag = new Selector(selectorData);
            frag.beginTransaction().add(R.id.selectorFrame, selectorFrag).commit();
        }

        //instantiates MAP
        Map mapFrag = (Map) frag.findFragmentById(R.id.mapFrame);
        if (mapFrag == null){
            mapFrag = new Map(mapGridData);
            frag.beginTransaction().add(R.id.mapFrame, mapFrag).commit();
        }
        //set selector fragment reference for map class
        mapFrag.setSelectorFrag(selectorFrag);

    }

}

File: Assignment2_PartA/app/src/main/java/curtin/edu/assignment2parta/MainActivity.kt
package curtin.edu.assignment2parta

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment

class MainActivity : AppCompatActivity() {

    lateinit var itemFragment : Fragment
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        supportActionBar?.hide()

        val moreFragment : Fragment = MoreFragment()
        itemFragment = ItemFragment.newInstance()

        val isFragmentContainerEmpty = savedInstanceState == null
        if(isFragmentContainerEmpty) {
            supportFragmentManager
                .beginTransaction()
                .add(R.id.item_fragment_container, itemFragment)
                .commit()
        }
        if(isFragmentContainerEmpty) {
            supportFragmentManager
                .beginTransaction()
                .add(R.id.more_fragmentContainerView, moreFragment, "moreFrag")
                .commit()
        }
    }

    //launches fragment on top half of screen
    fun startPostsFrag(id : Int) {
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.item_fragment_container, PostsFragment.newInstance(id), "postFrag")
            .commit()
    }
}